cmd/commands/cli/command.go

```diff
@@ -291,18 +291,29 @@ func (c *cliApp) healthcheck() {
 	info(buildString)
 }
 
-func (c *cliApp) proposals() {
+func (c *cliApp) proposals(filter string) {
 	proposals := c.fetchProposals()
 	c.fetchedProposals = proposals
-	info(fmt.Sprintf("Found %v proposals", len(proposals)))
+
+	filterMsg := ""
+	if filter != "" {
+		filterMsg = fmt.Sprintf("(filter: '%s')", filter)
+	}
+	info(fmt.Sprintf("Found %v proposals %s", len(proposals), filterMsg))
 
 	for _, proposal := range proposals {
 		country := proposal.ServiceDefinition.LocationOriginate.Country
 		if country == "" {
 			country = "Unknown"
 		}
+
 		msg := fmt.Sprintf("- provider id: %v, proposal id: %v, country: %v", proposal.ProviderID, proposal.ID, country)
-		info(msg)
+
+		if filter == "" ||
+			strings.Contains(proposal.ProviderID, filter) ||
+			strings.Contains(country, filter) {
```

Comment:
If it contains the only country, maybe `filter` is not a best name? Why not call it `country` or something similar?

Reply:
this filters not just for the country, but ProviderID itself, we might leave it 'filter'.

Reply:
true, missed it somehow, sorry.

Reply:
Other functions that accept an argument use `argString` for the argument identifier - I used 'filter' here to be more descriptive.